Microsoft.DotNet.Interactive/LanguageService/Range.cs

```diff
@@ -5,7 +5,13 @@ namespace Microsoft.DotNet.Interactive.LanguageService
 {
     public class Range
```

Comment:
What's the relationship of this to `TextSpan` (which we use a bunch in the `Parsing` namespace)? And how should we think about the split between types (like `HoverParam`) that are in `Microsoft.DotNet.Interactive.LanguageService` versus `Microsoft.DotNet.Interactive.App.Lsp`?

Reply:
While the types are similar, `TextSpan` is a Roslyn-specific type and I didn't want F# and PowerShell to be forced to use those types as well as force them to take a (indirect) dependency on Roslyn.  I also didn't want to leak Roslyn-specific stuff out into the language service layer which shouldn't know or care about Roslyn.

Reply:
The submission parser is using some types from `Microsoft.CodeAnalysis.Text` at the upper layer because they're robust, fast, and have a lot of functionality. We'll need *some* common abstraction at the polyglot syntax level so why not use these?

Reply:
That said, whatever that common abstraction is, let's consolidate it in `Microsoft.DotNet.Interactive.Parsing` and not up in the `dotnet-interactive` tool code.

Reply:
In that case I'll fall back to using `TextSpan` and only convert to/from LSP `Range` at the appropriate serialization layers.